src/validators/object.ts

```diff
@@ -64,11 +81,18 @@ export const object = <U extends Record<string, SomeSchema<any>>>(
         const validator = validators[key]!;
         const value = object[key];
         const r = validator.__validate(value);
-        isError ||= r._t === 'left';
+        if (r._t === 'left') {
+          errors.push({ path: key, error: r.value });
+          isError = true;
+        }
       }
 
       if (isError) {
-        return t.left(result as TypeOfResult);
+        return t.left({
+          expected: 'object',
```

Comment:
Wondering if we should use `this.toString()` instead of hardcoding a string?
If so, maybe we can remove `expected` here completely and let `refine(…)` do this?

Reply:
As long as this will be unique and easy to differentiate outside e.g. in function which is returning custom messages for users - I'm fine :)